tslib is missing as a dependency

Please use the following template to submit your issue. Following this template will allow us to quickly investigate and help you with your issue. Please be aware that issues which do not conform to this template may be closed.

For feature requests please contact us at team@intercom.io

## Version info

-   intercom-node version: 3.1.1
-   Node version: 16.13.2

## Expected behavior

intercom-node should install as a dependency and just work.
`tslib` listed as a dependency

## Actual behavior

intercom-node fails due to a missing dependency, `tslib`. Checking this modules package.json, it is listed as a dev dependency, but is required by the transpiled, distributed code.

## Steps to reproduce

1. `npm install intercom-client` into a new project
2. Write some code to use it.
3. 🔥

## Logs

```
yarn run v1.22.10
$ . .env && node src/index.js
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'tslib'
Require stack:
- /Users/anthonysceresini/code/propeller/intercom-integration-prototype/node_modules/intercom-client/dist/index.js
- /Users/anthonysceresini/code/propeller/intercom-integration-prototype/src/index.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/anthonysceresini/code/propeller/intercom-integration-prototype/node_modules/intercom-client/dist/index.js:4:15)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/anthonysceresini/code/propeller/intercom-integration-prototype/node_modules/intercom-client/dist/index.js',
    '/Users/anthonysceresini/code/propeller/intercom-integration-prototype/src/index.js'
  ]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

--- a/lib/contact.ts
+++ b/lib/contact.ts
@@ -19,6 +19,7 @@ export default class Contact {
     }
     createUser({
         externalId,
+        email,
         phone,
         name,
         avatar,
@@ -31,6 +32,7 @@ export default class Contact {
         const requestData: CreateContactRequest = {
             role: Role.USER,
             external_id: externalId,
+            email,
             phone,
             name,
             avatar,
@@ -55,7 +57,7 @@ export default class Contact {
             signed_up_at: data?.signedUpAt,
             last_seen_at: data?.lastSeenAt,
             owner_id: data?.ownerId,
-            unsubscribed_from_emails: data?.isUnsubscribedFromMails,
+            unsubscribed_from_emails: data?.isUnsubscribedFromEmails,
             custom_attributes: data?.customAttributes,
         };
         return this.client.post<ContactObject>({
@@ -185,8 +187,7 @@ type CreateContactRequest = Pick<ContactObject, 'role'> &
         >
     >;
 
-interface CreateUserData {
-    externalId?: CreateContactRequest['external_id'];
+interface CreateUserDataBase {
     phone?: CreateContactRequest['phone'];
     name?: CreateContactRequest['name'];
     avatar?: CreateContactRequest['avatar'];
@@ -197,16 +198,12 @@ interface CreateUserData {
     customAttributes?: CreateContactRequest['custom_attributes'];
 }
 
-interface CreateLeadData {
-    phone?: CreateContactRequest['phone'];
-    name?: CreateContactRequest['name'];
-    avatar?: CreateContactRequest['avatar'];
-    signedUpAt?: CreateContactRequest['signed_up_at'];
-    lastSeenAt?: CreateContactRequest['last_seen_at'];
-    ownerId?: CreateContactRequest['owner_id'];
-    isUnsubscribedFromMails?: CreateContactRequest['unsubscribed_from_emails'];
-    customAttributes?: CreateContactRequest['custom_attributes'];
+interface CreateUserData extends CreateUserDataBase {
+    email?: string;
+    externalId?: string;
 }
+
+type CreateLeadData = CreateUserDataBase;
 //
 interface RetrieveContactData {
     id: string;

--- a/lib/conversation.ts
+++ b/lib/conversation.ts
@@ -406,7 +406,7 @@ interface SnoozeConversationData {
 }
 //
 export enum CloseConversationMessageType {
-    CLOSED = 'closed',
+    CLOSED = 'close',
 }
 export enum CloseConversationType {
     ADMIN = 'admin',

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
     "name": "intercom-client",
-    "version": "3.1.1",
+    "version": "3.1.2",
     "description": "Official Node bindings to the Intercom API",
     "homepage": "https://github.com/intercom/intercom-node",
     "bugs:": "https://github.com/intercom/intercom-node/issues",

--- a/test/contact.ts
+++ b/test/contact.ts
@@ -5,7 +5,7 @@ import { Operators, Role } from '../lib/common/common.types';
 import { SearchContactOrderBy } from '../lib/contact';
 
 describe('contacts', () => {
-    it('should create a contact with user role', async () => {
+    it('should create a contact with user role with external id', async () => {
         const id = '536e564f316c83104c000020';
 
         const contact = {
@@ -32,6 +32,43 @@ describe('contacts', () => {
 
         const response = await client.contacts.createUser({
             externalId: contact.external_id,
+            phone: contact.phone,
+            name: contact.name,
+            avatar: contact.avatar,
+            signedUpAt: contact.signed_up_at,
+            lastSeenAt: contact.last_seen_at,
+            ownerId: contact.owner_id,
+            isUnsubscribedFromEmails: contact.unsubscribed_from_emails,
+        });
+
+        assert.deepStrictEqual(expectedReply, response);
+    });
+
+    it('should create a contact with user role with email', async () => {
+        const contact = {
+            role: 'user',
+            email: 'niko_bellic@mail.com',
+            phone: '+48370044567',
+            name: 'Niko Bellic',
+            avatar: 'https://nico-from-gta-iv.com/lets_go_bowling.jpg',
+            signed_up_at: 1638203719,
+            last_seen_at: 1638203719,
+            owner_id: 1,
+            unsubscribed_from_emails: true,
+        };
+
+        const expectedReply = {};
+
+        nock('https://api.intercom.io')
+            .post('/contacts', contact)
+            .reply(200, expectedReply);
+
+        const client = new Client({
+            usernameAuth: { username: 'foo', password: 'bar' },
+        });
+
+        const response = await client.contacts.createUser({
+            email: contact.email,
             phone: contact.phone,
             name: contact.name,
             avatar: contact.avatar,

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -7,7 +7,6 @@
         "target": "es5",
         "module": "commonjs",
         "moduleResolution": "node",
-        "importHelpers": true,
         "emitDecoratorMetadata": true,
         "experimentalDecorators": true,
         "sourceMap": true,

